packages/payment-processor/src/payment/any-to-erc20-proxy.ts

```diff
@@ -0,0 +1,124 @@
+ * @param feeAmount optionally, the fee amount to pay. Defaults to the fee amount.
+ * @param network optionally, network of the payment. Defaults to 'mainnet'
+ * @param overrides optionally, override default transaction values, like gas.
+ */
+export async function payAnyToErc20ProxyRequest(
+  request: ClientTypes.IRequestData,
+  paymentTokenAddress: string,
+  maxToSpend: BigNumberish,
+  signerOrProvider: Web3Provider | Signer = getProvider(),
+  amount?: BigNumberish,
+  feeAmount?: BigNumberish,
+  network: string = 'mainnet',
+  overrides?: ITransactionOverrides,
+): Promise<ContractTransaction> {
+  const encodedTx = await encodePayAnyToErc20ProxyRequest(
+    request,
+    paymentTokenAddress,
+    maxToSpend,
+    signerOrProvider,
+    amount,
+    feeAmount,
+    network,
+  );
+  const proxyAddress = proxyChainlinkConversionPath.getAddress(network);
+  const signer = getSigner(signerOrProvider);
+
+  const tx = await signer.sendTransaction({
+    data: encodedTx,
+    to: proxyAddress,
+    value: 0,
+    ...overrides,
+  });
+
+  return tx;
+}
+
+/**
+ * Encodes the call to pay a request with an ERC20 currency that is different from the request currency (eg. fiat). The payment is made by the ERC20 fee proxy contract.
+ * @param request request to pay
+ * @param paymentTokenAddress token address to pay with
+ * @param maxToSpend maximum of token the user is willing to spend
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ * @param feeAmountOverride optionally, the fee amount to pay. Defaults to the fee amount of the request.
+ * @param network optionally, network of the payment. Defaults to 'mainnet'
+ */
+export async function encodePayAnyToErc20ProxyRequest(
+  request: ClientTypes.IRequestData,
+  paymentTokenAddress: string,
+  maxToSpend: BigNumberish,
+  signerOrProvider: Web3Provider | Signer = getProvider(),
+  amount?: BigNumberish,
+  feeAmountOverride?: BigNumberish,
+  network: string = 'mainnet',
+): Promise<string> {
+  // get the conversion path
+  // Compute the path automatically
+  const paymentCurrency = { type: RequestLogicTypes.CURRENCY.ERC20, value: paymentTokenAddress };
+  const path = getConversionPath(request.currencyInfo, paymentCurrency, network);
+  if (!path) {
+    throw new Error(
+      `Impossible to find a conversion path between from ${request.currencyInfo} to ${paymentCurrency}`,
+    );
+  }
+
+  // Check request
+  validateConversionFeeProxyRequest(request, path, amount, feeAmountOverride);
+
+  const signer = getSigner(signerOrProvider);
+  const {
+    paymentReference,
+    paymentAddress,
+    feeAddress,
+    feeAmount,
+    maxRateTimespan,
+  } = getRequestPaymentValues(request);
+
+  const chainlinkDecimal = 8;
+  const decimalPadding = chainlinkDecimal - getDecimalsForCurrency(request.currencyInfo);
```

Comment:
isn't it possible that this is less than 0?

Reply:
It is, I have just tested. I will fix it and keep the test for later. Awesome finding.